### PR TITLE
integrate Facebook SDK

### PR DESCRIPTION
**Pull request by ace0625**
Added facebook SDK with the login button for android

### CODE REVIEW

File: src/client/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -26,6 +26,12 @@
         </intent-filter>
       </activity>
       <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
+        <meta-data android:name="com.facebook.sdk.ApplicationId" android:value="@string/facebook_app_id"/>
```
Comment (ethkim) [inline src/client/android/app/src/main/AndroidManifest.xml]: Didn't we have a limit on # of chars per each line? This is wrapping in github view. We should limit it to at least that # of chars..

File: src/client/android/app/src/main/java/com/yoda/MainActivity.java [hunk 2]
```diff
@@ -12,4 +15,10 @@
     protected String getMainComponentName() {
         return "yoda";
```
Comment (ethkim) [inline src/client/android/app/src/main/java/com/yoda/MainActivity.java]: What's the purpose of this method? There's gotta be a better way to construct this string. Perhaps declare it as constant at the top and use it here?

File: src/client/android/build.gradle [hunk 3]
```diff
@@ -18,7 +18,7 @@ allprojects {
         jcenter()
         maven {
             // All of React Native (JS, Obj-C sources, Android binaries) is installed
-            // from npm url "$rootDir/../node_modules/react-native/android"
+             url "$rootDir/../node_modules/react-native/android"
```
Comment (ethkim) [inline src/client/android/build.gradle]: indentation seems broken either here or the line above.

File: src/client/android/settings.gradle [hunk 4]
```diff
@@ -1,3 +1,5 @@
 rootProject.name = 'yoda'
 
 include ':app'
+include ':react-native-fbsdk'
+project(':react-native-fbsdk').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-fbsdk/android')
```
Comment (ethkim) [inline src/client/android/settings.gradle]: line break

File: src/client/javascripts/App.js [hunk 5]
```diff
@@ -35,19 +40,28 @@ class App extends Component {
...
-                                  <Text style={styles.instructions}>
-                      To get started, edit index.ios.js
-                                  </Text>
-                                  <Text style={styles.instructions}>
-                      Press Cmd+R to reload,                    {'\n'}
-                      Cmd+D or shake for dev menu
-                                  </Text>
-                          </View>
+
+              {/* Render facebook login button */}
```
Comment (ethkim) [inline src/client/javascripts/App.js]: indentation off

File: src/client/javascripts/App.js [hunk 6]
```diff
@@ -35,19 +40,28 @@ class App extends Component {
...
+             <LoginButton
+                onLoginFinished={
+                  (error, result) => {
+                    if (error) {
+                      alert('login has error: ' + result.error);
```
Comment (ethkim) [inline src/client/javascripts/App.js]: Where does this alert show? To the user?

In any case, capitalize Login and all user-visible / debug strings everywhere.

File: src/client/android/app/src/main/java/com/yoda/MainApplication.java [hunk 7]
```diff
@@ -1,19 +1,28 @@
...
 import com.facebook.react.ReactPackage;
 import com.facebook.react.shell.MainReactPackage;
+import com.facebook.reactnative.androidsdk.FBSDKPackage;
 
 import java.util.Arrays;
 import java.util.List;
 
 public class MainApplication extends Application implements ReactApplication {
 
+  // implement CallbackManager for facebook SDK
```
Comment (ethkim) [inline src/client/android/app/src/main/java/com/yoda/MainApplication.java]: Capitalize. Period at the end. 

"Implement CallbackManager for Facebook SDK."

**Comment by ethkim** [comment]
LGTM except the small suggestions above. Thanks!

File: src/client/android/app/src/main/java/com/yoda/MainActivity.java [hunk 8]
```diff
@@ -12,4 +15,10 @@
...
     }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
```
Comment (KaylaKwon) [inline src/client/android/app/src/main/java/com/yoda/MainActivity.java]: I think all of the tabs' length is 2 blanks

File: src/client/android/app/src/main/java/com/yoda/MainApplication.java [hunk 9]
```diff
@@ -23,7 +32,8 @@ protected boolean getUseDeveloperSupport() {
     @Override
     protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
-          new MainReactPackage()
+              new MainReactPackage(),
+              new FBSDKPackage(mCallbackManager)
```
Comment (KaylaKwon) [inline src/client/android/app/src/main/java/com/yoda/MainApplication.java]: is this 2 tabs?

Comment (ace0625) [inline src/client/android/app/src/main/java/com/yoda/MainApplication.java]: Yes. it seems right.

**Pull request closed without merging**